### ISSUE TITLE
0.32.2 Dashboard duplicate creates empty dashboard

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#9683]
Tried to reproduce issue from [the forum](https://discourse.metabase.com/t/copying-the-dashboard-failed/5834), by trying to duplicate a dashboard with a single card, which results in the new dashboard being created, but it's empty.

Metabase 0.32.1 JAR in OpenJDK with MariaDB 10.3 as metadata on Ubuntu 16.04

```
04-03 13:36:39 DEBUG middleware.log :: POST /api/dashboard 200 (51 ms) (4 DB calls). Jetty threads: 8/50 (3 busy, 5 idle, 0 queued) (77 total active threads)
04-03 13:36:39 WARN events.revision :: Failed to process revision event. :dashboard-create #error {
 :cause Assert failed: (db/exists? entity :id id)
 :via
 [{:type java.lang.AssertionError
   :message Assert failed: (db/exists? entity :id id)
   :at [metabase.models.revision$push_revision_BANG_ invokeStatic revision.clj 130]}]
 :trace
 [[metabase.models.revision$push_revision_BANG_ invokeStatic revision.clj 130]
  [metabase.models.revision$push_revision_BANG_ doInvoke revision.clj 130]
  [clojure.lang.RestFn invoke RestFn.java 1096]
  [metabase.events.revision$process_revision_event_BANG_ invokeStatic revision.clj 54]
  [metabase.events.revision$process_revision_event_BANG_ invoke revision.clj 36]
  [metabase.events$start_event_listener_BANG_$fn__12134$state_machine__6523__auto____12141$fn__12143 invoke events.clj 91]
  [metabase.events$start_event_listener_BANG_$fn__12134$state_machine__6523__auto____12141 invoke events.clj 91]
  [clojure.core.async.impl.ioc_macros$run_state_machine invokeStatic ioc_macros.clj 973]
  [clojure.core.async.impl.ioc_macros$run_state_machine invoke ioc_macros.clj 972]
  [clojure.core.async.impl.ioc_macros$run_state_machine_wrapped invokeStatic ioc_macros.clj 977]
  [clojure.core.async.impl.ioc_macros$run_state_machine_wrapped invoke ioc_macros.clj 975]
  [clojure.core.async.impl.ioc_macros$take_BANG_$fn__6541 invoke ioc_macros.clj 986]
  [clojure.core.async.impl.channels.ManyToManyChannel$fn__1443 invoke channels.clj 135]
  [clojure.lang.AFn run AFn.java 22]
  [java.util.concurrent.ThreadPoolExecutor runWorker ThreadPoolExecutor.java 1149]
  [java.util.concurrent.ThreadPoolExecutor$Worker run ThreadPoolExecutor.java 624]
  [java.lang.Thread run Thread.java 748]]}
```

⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by botchniaque** [comment]
I am also seeing this porblem in both `v0.32.1` and `v0.32.2`. It works fine in `v0.31.2`.

**Comment by camsaul** [comment]
@tlrobinson @kdoh I've looked into this for a while. There are bugs in the new `DashboardCopyModal` and `EntityCopyModal` components. The problem is that the FE client is dispatching a `metabase/entities/dashboards/CREATE` action instead of the `COPY` action that one would expect.

It looks like what's happening is that the `bindActionCreators` stuff  [here](https://github.com/metabase/metabase/blob/v0.32.2/frontend/src/metabase/entities/containers/EntityType.jsx#L37) in the `entityType` decorator on `EntityForm` is generating the `_boundActionCreators` map with `{create: Dashboards.actions.create}`, and this ultimately gets passed as a prop to `EntityForm` [here](https://github.com/metabase/metabase/blob/v0.32.2/frontend/src/metabase/entities/containers/EntityType.jsx#L45). This overrides the custom value of `create` (to do copying) you're setting [here](https://github.com/metabase/metabase/blob/v0.32.2/frontend/src/metabase/entities/containers/EntityCopyModal.jsx#L23). 

I tried reverting your change [here](https://github.com/metabase/metabase/commit/db87e11e1a3c5c7552e9cd5cddbb930d76b7ba1d#diff-9ecbbf5169e983aa9944dbd2f0031845) to see if it would fix it but  

```js
<ComposedComponent {...this._boundActionCreators} {...this.props} />
```

would fix it but I get a whole new error.

```
DashboardCopyModal.jsx:50 Uncaught (in promise) TypeError: Cannot read property 'id' of undefined
    at onSaved (DashboardCopyModal.jsx:50)
    at onSubmitSuccess (EntityForm.jsx:35)
    at handleSubmit.js:37
```

`EntityForm` is looking for `action.payload.object` [here](https://github.com/metabase/metabase/blob/v0.32.2/frontend/src/metabase/entities/containers/EntityForm.jsx#L33) but instead `action` is something like

```js
{
  type: "metabase/entities/dashboards/COPY",
  payload: {
   entities: {
      dashboards: {
        453: {…} // new dashboard
      }
    },
    result: 453
  }
}
```

It looks to me like this change [here](https://github.com/metabase/metabase/commit/bb4c7c8661d3f78565013d026532ce20db60a3dc#diff-753fb05b0a040230d89166f8c0a5c1d7R33) might have broke the second thing going on here. Not sure what changes should be made to fix Dashboard copying without breaking the things that those changes were meant to fix.

In the future is there some way we can prevent these sorts of regressions? It seems to me a E2E test for this would be pretty simple in theory (Have Dashboard w/ 1 Card; click Copy button, click "Duplicate" -- see what happens)

**Comment by tlrobinson** [comment]
I believe PR #9524 fixes this. Not sure why I didn’t get that merged before 0.32 :-/

**Comment by camsaul** [comment]
Fixed by #9524
